HPCC-14453 Preflight upgrade

miguelvazq opened the pull request:
Add pre flight submission form into new ECL Watch.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Automated Smoketest 
Sha: b316a1645e4b7fd65026bb257842a869e990c8f7
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/PreflightDetailsWidget.js:
```diff
@@ -0,0 +1,247 @@
+                    var clean = finalColumns[index].replace(/([~!@#$%^&*()_+=`{}\[\]\|\\:;'<>,.\/? ])+/g, '').replace(/^(-)+|(-)+$/g,'');
+                    if (clean === 'Condition') {
+                        lang.mixin(dynamicColumns, {
+                            [clean] : {
```
What does this syntax do (I have not seen it before)?

miguelvazq replied:
This creates dynamic properties with the value from clean which in turn is 'Condition' 'State' 'RoxieCluster'

GordonSmith replied:
I didn't know you could do that with this syntax!

GordonSmith commented on esp/src/eclwatch/PreflightDetailsWidget.js:
```diff
@@ -0,0 +1,247 @@
+            if (params) {
+                var finalColumns = params.Columns.Item;
+                for (index in finalColumns) {
+                    var clean = finalColumns[index].replace(/([~!@#$%^&*()_+=`{}\[\]\|\\:;'<>,.\/? ])+/g, '').replace(/^(-)+|(-)+$/g,'');
```
Is this removing HTML formatting?

miguelvazq replied:
Since the response returns "Column Name" I am removing any spaces and or special characters in it so that they come back like so: "ColumnName"

GordonSmith commented on esp/src/eclwatch/PreflightDetailsWidget.js:
```diff
@@ -0,0 +1,247 @@
+                                }
+                            }
+                        });
+                    } if (clean === 'State') {
```
The "if" should be on a new line.

GordonSmith commented on esp/src/eclwatch/TopologyWidget.js:
```diff
@@ -18,53 +18,163 @@ define([
+                            WsMachine.GetMachineInfo ({
+                                request: {
+                                    [MachineInformationClean]: filter[MachineInformationClean],
+                                    [MachineInformationCount]: 1000,
```
Is this syntax correct?

miguelvazq replied:
I verified with @wangkx this count seems to increase based on selections and is subjective.

GordonSmith commented on esp/src/eclwatch/TopologyWidget.js:
```diff
@@ -18,53 +18,163 @@ define([
+                        WsMachine.GetTargetClusterInfo ({
+                            request: {
+                                [TargetClustersClean]: filter[TargetClustersClean],
+                                [TargetClusterCount]: 1000,
```
Is this syntax correct?

miguelvazq replied:
I verified with @wangkx this count seems to increase based on selections and is subjective.

GordonSmith commented on esp/src/eclwatch/TopologyWidget.js:
```diff
@@ -159,13 +323,165 @@ define([
+        },
+
+        createLabelAndElement: function (id, label, element, placeholder, value) {
+            var context = this;
```
should declare ```var element;``` here and then assign values to it in the switch statement

GordonSmith commented:
Some questions.

miguelvazq commented:
@GordonSmith updated.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 52dd2a39bf11fc437fc73f65156ec8c3bf958300
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@miguelvazq Needs rebasing for a clean merge

miguelvazq commented:
@richardkchapman rebased.

HPCCSmoketest commented:
Automated Smoketest 
Sha: 684916f238f8de5c3d4b5b967bc733b188d754b5
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

Commit:
HPCC-14453 Preflight upgrade

Add pre flight submission form into new ECL Watch.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
@GordonSmith changes made after latest comments

HPCCSmoketest commented:
Automated Smoketest 
Sha: 37283ae0c8663cb2e57db8814e7aeea57a7585b4
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
Unittest result: total:83 passed:83 failed:0

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
@GordonSmith Please re-review/approve

GordonSmith commented:
Good to go.
